src/samples/aspnet/Elsa.Samples.CustomTenantIdSource/CustomTenantAccessor.cs

```diff
@@ -0,0 +1,29 @@
+using Elsa.Services;
+using Microsoft.AspNetCore.Http;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Elsa.Samples.CustomTenantIdSource
+{
+    public class CustomTenantAccessor : ITenantAccessor
+    {
+        private readonly IHttpContextAccessor _accessor;
+
+        public CustomTenantAccessor(IHttpContextAccessor accessor)
+        {
+            _accessor = accessor;
+        }
+        public Task<string> GetTenantIdAsync(CancellationToken cancellationToken = default)
+        {
+            var id = "elsa-core";
+
+            //You can customize the data
+            var httpContext = _accessor.HttpContext;
+
+            return Task.FromResult(id);
```

Comment:
Since you injected `IHttpContextAccessor`, how about actually reading the tenant ID from the HTTP context? Could be query string or HTTP header. For example:

```csharp
var httpContext = _accessor.HttpContext;
var tenantId = httpContext.Request.Headers["x-tenant"]; 

return Task.FromResult(tenantId);
```

This would make for a more real-world example I think.
Otherwise, you may just as well not inject `IHttpContextAccessor` and return "elsa-core" directly as the tenant ID.